File: 프로그래머스/1/12921. 소수 찾기/소수 찾기.java
class Solution {
    public int solution(int n) {
        // 2는 소수이므로 초기값 1로 선언
        // 이후 2부터 n의 제곱근까지 나눠주면서 소수 판별할거라 2는 따로 세어줌
        int answer = 1;
        // 3부터 n까지 검사해주기
        for(int i=3; i<=n; i++){
            if(isPrime(i)){
                answer+=1;
            }
        }
        
        return answer;
    }
    
    // 소수인지 판별해주는 함수
    private boolean isPrime(int num){
        // 2부터 n의 제곱근까지 반복하여 나누어지면 소수 아님
        for(int i =2; i<=(int)Math.sqrt(num); i++){
            if(num%i==0){
                return false;
            }
        }
        return true;
    }
}

File: 프로그래머스/2/42839. 소수 찾기/소수 찾기.py

def isPrime(number):
    if number < 2:
        return False
    for i in range(2, int(number**0.5)+1):
        if number % i == 0:
            return False
    return True



    
def solution(numbers):

    visited = [False] * len(numbers)
    num_set = set()
    sequence = [""] * len(numbers)
    def backtrack(depth):
        if depth > 0:
            num = int(''.join(sequence[:depth]))
            if isPrime(num):
                num_set.add(num)
        for i in range(len(numbers)):
            if not visited[i]:
                visited[i] = True
                sequence[depth] = numbers[i]
                backtrack(depth+1)
                visited[i] = False
    
    backtrack(0)
    return len(num_set)